core/block_validator.go

```diff
@@ -114,25 +96,11 @@ func (v *BlockValidator) ValidateBody(block *types.Block) error {
 	if err := v.ValidateL1Messages(block); err != nil {
 		return err
 	}
-	if v.checkCircuitCapacity {
-		// if a block's RowConsumption has been stored, which means it has been processed before,
-		// (e.g., in miner/worker.go or in insertChain),
-		// we simply skip its calculation and validation
-		if rawdb.ReadBlockRowConsumption(v.bc.db, block.Hash()) != nil {
-			return nil
-		}
-		rowConsumption, err := v.validateCircuitRowConsumption(block)
-		if err != nil {
+
+	if v.asyncValidator != nil {
+		if err := v.asyncValidator(block); err != nil {
```

Comment:
Might be interesting to add a time metric here (or inside `Check`), to see how often and for how long this call is blocking.

Reply:
I had something similar in worker but not here, thanks! added in https://github.com/scroll-tech/go-ethereum/pull/996/commits/0a8132aa7aee795f64f5bf0de54c25849ac752f1